### PR TITLE
Add tests for interop between source and prebuilt extensions

### LINKED ISSUE #9333: Add test for prebuilt and source extension packages that depend on each other
I think we should add a usage test that tests that a prebuilt extension can depend on a source extension and vice versa. If only because I keep messing it up :(.

### PR DESCRIPTION
**Pull request by afshin**
Adds some tests and a workflow job for the interoperation between source and prebuilt extensions.

## References

Fixes #9333

## Code changes

* Source extension depending on a token provided by a prebuilt extension
* Prebuilt extension depending on a token provided by a source extension
* Prebuilt extension shadowing a source extension

## User-facing changes
N/A

## Backwards-incompatible changes
N/A

### CODE REVIEW

File: jupyterlab/tests/mock_packages/interop/consumer/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "version": "3.0.0-rc.13",
+  "private": true,
+  "dependencies": {
+    "@jupyterlab/mock-token": "^3.0.0-rc.13"
+  },
+  "devDependencies": {
+    "@jupyterlab/builder": "^3.0.0-rc.13"
+  },
+  "jupyterlab": {
+    "extension": true
```
Comment (jasongrout) [inline jupyterlab/tests/mock_packages/interop/consumer/package.json:12]:
```suggestion
    "extension": true,
    "sharedPackages": {
      "@jupyterlab/mock-token": {
        singleton: true,
        bundled: false
      }
    }
```

File: jupyterlab/tests/mock_packages/interop/consumer/setup.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+js_name = data['name']
+
+setup(name=name,
+      version=data['version'],
+      py_modules = [name],
+      data_files = [
+        (f'share/jupyter/labextensions/{js_name}', glob('static/package.json')),
+        (f'share/jupyter/labextensions/{js_name}', glob('static/static/*'))
+    ])
```
Comment (jasongrout) [inline jupyterlab/tests/mock_packages/interop/consumer/setup.py:21]:
```suggestion
    ])

```

File: jupyterlab/tests/mock_packages/interop/consumer/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "devDependencies": {
+    "@jupyterlab/builder": "^3.0.0-rc.13"
+  },
+  "jupyterlab": {
+    "extension": true,
+    "sharedPackages": {
+      "@jupyterlab/mock-token": {
+        singleton: true,
+        bundled: false
```
Comment (jtpio) [inline jupyterlab/tests/mock_packages/interop/consumer/package.json:16]: Missing quotes?

```suggestion
        "singleton": true,
        "bundled": false
```

**Review by blink1073 — APPROVED** [review]
Thanks!

**Pull request merged**